dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/Compilation.java

```diff
@@ -80,7 +81,7 @@ private int compileThread(Thread thread, int nextId) {
         pred.setCId(nextId++);
 
         while (toBeCompiled != null) {
-            List<Event> compiledEvents = toBeCompiled.compile(target);
+            List<Event> compiledEvents = toBeCompiled.accept(new CompilationVisitor(target));
```

Comment:
The `CompilationVisitor` has immutable state, so just create it once and reuse it for all calls.